portfolio/src/main/webapp/jquery_functions.js

```diff
@@ -1,31 +1,59 @@
 /*
 * Purpose: contains various JQuery functions that listen for events on main portfolio page
 */
+// For each individual background img, width is 2016px height is 1512px
+const BACKGROUND_IMG_WIDTH = 2016;
+const BACKGROUND_IMG_HEIGHT = 1512;
+const VERT_PADDING = 40;
+const HORZ_PADDING = 20;
+const SPRITE_IMG_ROWS = 2;
+const SPRITE_IMG_COLS = 3;
+const NUM_IMAGES = 5;
 
-let img_arr = new Array(5);
-img_arr = ["../images/mountbierstadt.jpg", 
-           "../images/mountyale.jpg",
-           "../images/gardenofgods.jpg",
-           "../images/lakedillon.jpg",
-           "../images/mountelbert.jpg"
-           ];
 labels = ["the summit of Mount Bierdstadt.", 
           "the summit of Mount Yale.", 
+          "the summit of Mount Elbert, the tallest point in Colorado.",
           "the Garden of the Gods.", 
-          "Lake Dillon, seen from the vantage point of driving along Dillon Dam road.",
-          "the summit of Mount Elbert, the tallest point in Colorado."];
+          "Lake Dillon, seen from the vantage point of driving along Dillon Dam road."];
 
+profile_img_paths = ["images/noogler.gif",
+                     "images/zoebakercircle.png"
+                    ];
+
+function preload_imgs(img_arr, img_paths){
+  for (let i = 0; i < img_paths.length; i++) {
+      img_arr[i] = new Image();
+      img_arr[i].src = img_paths[i];
+  }
+}
 
 $(document).ready(function(){
+    // preload profile icon images + gif
+    preload_imgs(new Array(2), profile_img_paths);
```

Comment:
How is `new Array(2)` accessed once all the images are placed there within?

Reply:
The original point of the array was supposed to be arbitrary, basically just chuck things into an array so that the website is forced to immediately preload the images as soon as the site loads. 
I changed it so that the function actually returns the array and hover on icon now switches between array[0] and array[1].